### ISSUE TITLE
Wrong .so file name for implicit package.

### ISSUE THREAD

**Issue by leojay** [Nuitka/Nuitka#154]
I'm using Python 3.5.3, Nuitka 0.6.0 in a venv environment on a linux machine.
There is only one file in my project and here is my project structure:
```
package_test/
    util.py
```
The content of package_test/util.py is
```
$ cat package_test/util.py
def add(a, b):
    return a + b
```

Since Python 3.3, `__init__.py` is not needed anymore for a package. I can import and run the code in util.py without any problems. e.g.
```
$ python -c 'from package_test.util import add; print(add(1,2))'
3
```

When I run Nuitka, here is what I get:
```
$ python -m nuitka --module package_test --include-package=package_test
Nuitka:WARNING:Recursed to package 'package_test' at '/home/leojay/test/py3test/src/package_test' twice.
$ ls -l -a
total 188
drwxr-xr-x 4 leojay leojay   4096 Sep 28 15:46 .
drwxr-xr-x 7 leojay leojay   4096 Sep 28 13:59 ..
drwxr-xr-x 3 leojay leojay   4096 Sep 28 15:46 .build
drwxr-xr-x 3 leojay leojay   4096 Sep 28 15:41 package_test
-rw-r--r-- 1 leojay leojay    673 Sep 28 15:46 .pyi
-rw-r--r-- 1 leojay leojay 170648 Sep 28 15:46 .so
```

You can see the .so file has no base name. The full name is just ".so", while I'm expecting package_test.so

And as long as I run `touch package_test/__init__.py` and run Nuitka again, I will get correct ".so" file name.
```
$ rm -rf .build/ .pyi .so
$ touch package_test/__init__.py
$ python -m nuitka --module package_test --include-package=package_test
Nuitka:WARNING:Recursed to package 'package_test' at '/home/leojay/test/py3test/src/package_test' twice.
$ ls -l -a
total 188
drwxr-xr-x 4 leojay leojay   4096 Sep 28 15:54 .
drwxr-xr-x 7 leojay leojay   4096 Sep 28 13:59 ..
drwxr-xr-x 3 leojay leojay   4096 Sep 28 15:49 package_test
drwxr-xr-x 3 leojay leojay   4096 Sep 28 15:54 package_test.build
-rw-r--r-- 1 leojay leojay    640 Sep 28 15:54 package_test.pyi
-rw-r--r-- 1 leojay leojay 170680 Sep 28 15:54 package_test.so
```

So I have 2 questions:
1. Can you please make Nuikta support the case where there is no `__init__.py` file in package?
2. Why the "Recursed to package twice" warning? How do I get rid of it?

Thanks!
Leo

**Comment by kayhayen** [comment]
Nice report, thank you. For these so called "name space packages" Nuitka only so far has been applied to ones inside programs or for sub-packages, but as you point out, that is wrong and easy to fix. I will include it in the 0.6.0.1 hotfix later this weekend.

Funnily the files become hidden due to the base name being empty. I imagine people could also easily have believed there was no output.

As for the warning, using the --include-package is a kludge and work-around until including the package becomes automatic, which in turn depends on adding name based variants. I expect it to be changed during the 0.6.x series eventually, by adding a ``--module-by-name`` and ``--module-by-filename`` distinction, after which the warning would actually be useful.

For the time being it's annoying and apparently disables you from using --nofollow-import-to.

Yours,
Kay

**Comment by kayhayen** [comment]
Released as part of 0.6.0.1
